src/Workspaces/CSharp/Portable/Execution/CSharpOptionsSerializationService.cs

```diff
@@ -32,6 +34,15 @@ public override void WriteTo(ParseOptions options, ObjectWriter writer, Cancella
             writer.WriteValue(options.PreprocessorSymbolNames.ToArray());
         }
 
+        public override void WriteTo(OptionSet options, ObjectWriter writer, CancellationToken cancellationToken)
+        {
+            WriteOptionSetTo(options, LanguageNames.CSharp, writer, cancellationToken);
+
+            WriteOptionTo(options, CSharpCodeStyleOptions.UseImplicitTypeForIntrinsicTypes, writer, cancellationToken);
```

Comment:
This seems like a lot of work to go and add all necessary options into this when we add new options, or need to remote options.  Can we not just enumerate all the optoins and send them along?